### PR TITLE
[Logging] Document the Monolog `reset()` method for long running processes

### LINKED ISSUE #15086: [Logging] Please advise people to use reset() for long running processes
See https://twitter.com/seldaek/status/1370115718499422208 for details.

The latest recipe [has a buffer limit now](https://github.com/symfony/recipes/blob/afacc960726feea45134a9b76634e3b80ab818ad/symfony/monolog-bundle/3.1/config/packages/prod/monolog.yaml#L7) so it's less of an issue I guess, but still reset is cleaner.

### PR DESCRIPTION
**Pull request by 94noni**
Close https://github.com/symfony/symfony-docs/issues/15086

### CODE REVIEW

**Comment by GromNaN** [comment]
`buffer_size` for finger_crossed can be mentioned.
https://github.com/symfony/recipes/blob/master/symfony/monolog-bundle/3.7/config/packages/monolog.yaml#L48

File: logging.rst [hunk 1]
```diff
@@ -367,6 +367,13 @@ information to your log entries.
 
 See :doc:`/logging/processors` for details.
 
+Handling logs in long running processes
+---------------------------------------
+
+During long running processes, logs can be accumulated into Monolog and cause some buffer overflow,
+memory increase or even non logical logs.
+Monolog supports to be reseted via the ``reset()`` function, this allows to empty the log pool.
```
Comment (GromNaN) [inline logging.rst]:
```suggestion
Monolog in-memory data can be cleared using the ``reset()`` function.
```

File: logging.rst [hunk 2]
```diff
@@ -367,6 +367,13 @@ information to your log entries.
 
 See :doc:`/logging/processors` for details.
 
+Handling logs in long running processes
+---------------------------------------
+
+During long running processes, logs can be accumulated into Monolog and cause some buffer overflow,
+memory increase or even non logical logs.
+Monolog in-memory data can be cleared using the ``reset()`` function.
```
Comment (Seldaek) [inline logging.rst]:
```suggestion
Monolog in-memory data can be cleared using the ``reset()`` method on a ``Monolog\Logger`` instance.
This should typically be called between every jobs/tasks that a long running process is working through.
```

**Comment by kbond** [comment]
To clarify, this is handled automatically when consuming messenger messages, correct?

**Comment by Seldaek** [comment]
Yes I believe so as of https://github.com/symfony/symfony/pull/40761

**Comment by 94noni** [comment]
@GromNaN as early reviewer, do you see something else to add on this PR? thank you

**Comment by GromNaN** [comment]
All good. I'm wondering why this is not automatic.

**Comment by OskarStark** [comment]
Should be against `4.4`

**Comment by 94noni** [comment]
@OskarStark fair enough, will handle it soon :)
thx @GromNaN for answer

**Comment by 94noni** [comment]
v4 is eol, thus this PR is reviewable as it is

**Comment by OskarStark** [comment]
Thanks Antoine.

**Pull request merged**